src/master-playlist-controller.js

```diff
@@ -1574,12 +1049,18 @@ export class MasterPlaylistController extends videojs.EventTarget {
     this.masterPlaylistLoader_.dispose();
     this.mainSegmentLoader_.dispose();
 
-    if (this.audioPlaylistLoader_) {
-      this.audioPlaylistLoader_.dispose();
-    }
-    if (this.subtitlePlaylistLoader_) {
-      this.subtitlePlaylistLoader_.dispose();
-    }
+    ['AUDIO', 'SUBTITLES'].forEach((type) => {
+      const groups = this.mediaGroups_[type].groups;
```

Comment:
I agree the groups groups group grouping gets confusing. I wasn't sure what the best naming would be since as you said, there isn't a collective term, but I like `mediaTypes` I'll note that the m3u8-parser outputs a `mediaGroups` property on the parsed manifest object. Not sure if we would rather come up with a naming convention that ties the two together